Mage.Sets/src/mage/cards/t/TheFifteenthDoctor.java

```diff
@@ -0,0 +1,160 @@
+package mage.cards.t;
+
+import mage.MageInt;
+import mage.abilities.Ability;
+import mage.abilities.common.EntersBattlefieldOrAttacksSourceTriggeredAbility;
+import mage.abilities.common.SimpleStaticAbility;
+import mage.abilities.effects.OneShotEffect;
+import mage.abilities.effects.common.continuous.GainAbilityControlledSpellsEffect;
+import mage.abilities.keyword.ImproviseAbility;
+import mage.cards.*;
+import mage.constants.*;
+import mage.filter.FilterCard;
+import mage.filter.common.FilterArtifactCard;
+import mage.filter.common.FilterNonlandCard;
+import mage.filter.predicate.ObjectSourcePlayer;
+import mage.filter.predicate.ObjectSourcePlayerPredicate;
+import mage.filter.predicate.Predicates;
+import mage.filter.predicate.mageobject.ManaValuePredicate;
+import mage.game.Game;
+import mage.game.events.GameEvent;
+import mage.game.stack.Spell;
+import mage.players.Player;
+import mage.target.TargetCard;
+import mage.watchers.Watcher;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+
+/**
+ * @author balazskristof
+ */
+public class TheFifteenthDoctor extends CardImpl {
+
+    private static final FilterNonlandCard filter = new FilterNonlandCard("the first nonartifact spell you cast each turn");
+
+    static {
+        filter.add(TheFifteenthDoctorCastSpellPredicate.instance);
+    }
+
+    public TheFifteenthDoctor(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.CREATURE}, "{2}{U}{R}");
+
+        this.supertype.add(SuperType.LEGENDARY);
+        this.subtype.add(SubType.TIME_LORD);
+        this.subtype.add(SubType.DOCTOR);
+        this.power = new MageInt(3);
+        this.toughness = new MageInt(3);
+
+        // Whenever The Fifteenth Doctor enters or attacks, mill three cards. You may put an artifact card with mana value 2 or 3 from among them into your hand.
+        this.addAbility(new EntersBattlefieldOrAttacksSourceTriggeredAbility(new TheFifteenthDoctorEffect(), false));
+
+        // The first nonartifact spell you cast each turn has improvise. (Your artifacts can help cast that spell. Each artifact you tap after you’re done activating mana abilities pays for {1}.)
+        this.addAbility(new SimpleStaticAbility(
+                new GainAbilityControlledSpellsEffect(new ImproviseAbility(), filter)
+                    .setText("The first nonartifact spell you cast each turn has improvise. " +
+                            "<i>(Your artifacts can help cast that spell. Each artifact you " +
+                            "tap after you’re done activating mana abilities pays for {1}.)")),
```

Comment:
Error: (rules) card's rules contains restricted symbol ’ for SLD - The Fifteenth Doctor - 1584